lib/dotenv/beefy/railtie.rb

```diff
@@ -6,15 +6,37 @@ module Beefy
     class Railtie < Rails::Railtie
       config.before_configuration { load_environments }
 
+      # Load environment dotfiles in the following order (e.g. if in "test" environment on "darwin")
+      # 1. .env.test.local
+      # 2. .env.darwin.local
+      # 3. .env.local
+      # 4. .env.test
+      # 5. .env.darwin
+      # 6. .env
+      # 
+      # The order matters, because the files loaded first will "lock in" the value for that ENV var.
+      # Dotenv.load memoizes each ENV value, and if the value is set, it cannot be updated later.
+      # If you want to update ENV values as new values come in, you need to use Dotenv.overload(*files)
       def load_environments
-        files = environments.map { |env| ".env.#{env}" }
+        files = []
+        
+        environments.each do |env| 
+          files << ".env.#{env}.local"
+        end
+
+        files << '.env.local' unless Rails.env.test? # this is a dotenv-rails convention
```

Comment:
minor: it might help to point to the source code in `dotenv` for this

Reply:
oh, you're right! I linked it in a few others places, but not the code comment. 👨🏻‍🎓